coral/src/app/pages/index.test.tsx

```diff
@@ -1,15 +1,58 @@
 import HomePage from "src/app/pages";
 import { render, cleanup, screen } from "@testing-library/react";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactElement } from "react";
+
+export function renderWithClient(client: QueryClient, ui: ReactElement) {
```

Comment:
I think we will use similar pattern in the future in most components that use server state. Could we already consider refactoring this into a some utility. Also, we could utilize the [wrapper](https://testing-library.com/docs/react-testing-library/api/#wrapper) from RTL.

We could also consider adding a import restriction that no component test should import those application internals, but we can do that in a followup PR (if ever).

Reply:
Sounds good to me! Very good case for a utils test function here (I'm normally a bit wary of them, but this makes total sense). 

Also agree to implement a restricted import rule separate if we want it later!